backend/persistent/datastores/drivers/AWSAthenaDriver.js

```diff
@@ -0,0 +1,222 @@
+            }
+            return resolve(queryState);
+        });
+    });
+}
+
+/**
+ * The following method will stop the query execution based on the query id
+ * @param {object} athenaClient  - Object created using create athena client
+ * @param {string} queryExecutionId - AWS Athena Query Id
+ * @returns {Promise} That resolves to AWS Stop Request
+ */
+export function stopQuery(athenaClient, queryExecutionId) {
+    const client = athenaClient;
+
+    const queryParams = {
+        QueryExecutionId: queryExecutionId
+    };
+
+    return new Promise(function(resolve, reject) {
+        return client.stopQueryExecution(queryParams, (err, data) => {
+            if (err) {
+                Logger.log(`Unexpected Error stoping Athena Query Execution ${err}`);
+                return reject(err);
+            }
+            return resolve(data);
+        });
+    });
+}
+
+/**
+ * The following method will get the query results based on the query id
+ * @param {object} athenaClient  - Object created using create athena client
+ * @param {string} queryExecutionId - AWS Athena Query Id
+ * @returns {Promise} Resolves to AWS Query Response
+ */
+export function getQueryResults(athenaClient, queryExecutionId) {
+    const client = athenaClient;
+
+    const queryParams = {
+        QueryExecutionId: queryExecutionId
+    };
+
+    return new Promise(function(resolve, reject) {
+        client.getQueryResults(queryParams, (err, data) => {
+            if (err) {
+                return reject(err);
+            }
+            return resolve(data);
+        });
+    });
+}
+
+/**
+ * The following function will execute a query against athena.  It will first start
+ * by starting the query request.  It will then start a timer to periodically
+ * check to see if the query results have completed. If the Query Timeout has
+ * Exceeded it will reject the query.  If the it receives data before the query
+ * times out it will return the response.
+ * @param {object} queryParams - Query Parameters
+ * @param {string} queryParams.accessKey - AWS Access Key
+ * @param {string} queryParams.secretAccessKey - AWS Secret Key
+ * @param {string} queryParams.region - AWS Region
+ * @param {string} queryParams.region - AWS Region
+ * @param {string} queryParams.dbName - Database name to connect to
+ * @param {string} queryParams.sqlStatement - SQL statement to execute
+ * @param {string} queryParams.s3Outputlocation - Location will Athena will output resutls of query
+ * @param {number} queryParams.queryTimeout  - The timeout interval when the query should stop
+ * @returns {Promise} resolve to AWS Query Response
+ */
+export function executeQuery(queryParams) {
+    const client = createAthenaClient(queryParams);
+
+    return new Promise(function(resolve, reject) {
+        return startQuery(client, queryParams).then(queryExecutionId => {
+
+            // Define the wait interval
+            let retryInterval = queryParams.queryTimeout / NUMBER_OF_RETRIES;
```

Comment:
@n-riesco I have been working my way through you comments and wanted to discuss the polling strategy.  I think we should have one approach.  It should either poll on a fixed interval (i.e. Polling Interval is 2000 ms and it tries every 2 seconds ) or based on a query timeout which has a fixed number of retries ( 2 minute timeout and tries every 24 seconds).  I don't have a preference.  Which way would you like to proceed?